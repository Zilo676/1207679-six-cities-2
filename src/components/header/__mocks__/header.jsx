import React from "react";

const Header = () => {
  return <></>;
};

export {Header};
export default Header;
