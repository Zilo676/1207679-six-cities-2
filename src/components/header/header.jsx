import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

import {getUser} from '../../reducer/user/selectors';

const Header = (props) => {
  const email = Object.keys(props.user).length === 0 ? `Sign in` : props.user.email;
  const avatarUrl = Object.keys(props.user).length === 0 ? `` : props.user[`avatar_url`];
  const link = Object.keys(props.user).length === 0 ? `/login` : `/favorites`;

  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <a className="header__logo-link header__logo-link--active">
              <img className="header__logo" src="/img/logo.svg" alt="img/logo.svg" width="81" height="41" />
            </a>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <Link to={link} className="header__nav-link header__nav-link--profile">
                  <div className="header__avatar-wrapper user__avatar-wrapper">
                    <img className="user__avatar" src={`https://htmlacademy-react-2.appspot.com/six-cities` + avatarUrl} alt="" width="81" height="41" />
                  </div>
                  <span className="header__user-name user__name">{email}</span>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  user: PropTypes.shape().isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  user: getUser(state),
});

export {Header};

export default connect(mapStateToProps, null)(Header);
