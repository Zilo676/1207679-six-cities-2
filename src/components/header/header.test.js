import React from 'react';
import renderer from 'react-test-renderer';

import {Header} from './header.jsx';

const user = {
  id: 1,
  email: `Sign In`,
  name: `Sign IN`,
  [`avatar_url`]: ``,
  [`is_pro`]: false
};

it(`Component is rendered correctly`, () => {
  const tree = renderer.create(<Header
    user={user}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
