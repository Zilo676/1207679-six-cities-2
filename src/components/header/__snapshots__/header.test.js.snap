// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is rendered correctly 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="/img/logo.svg"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt=""
                  className="user__avatar"
                  height="41"
                  src="https://htmlacademy-react-2.appspot.com/six-citiesundefined"
                  width="81"
                />
              </div>
              <span
                className="header__user-name user__name"
              >
                Sign In
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
